### PR TITLE
add options map to graphql-network

### PR DESCRIPTION
**Pull request by MageMasher**
This will allow custom auth headers to be supplied in a graphql request.

### CODE REVIEW

File: src/com/wsscode/pathom/fulcro/network.cljs [hunk 1]
```diff
@@ -187,24 +187,29 @@
...
+        (fn-network
+          (fn [this edn ok error]
+              (go
+                (try
+                  (let [edn (-> edn
+                                p/query->ast
+                                (p/elide-ast-nodes #{::pp/profile})
+                                p/ast->query)
+                        query (pg/query->graphql edn {::pg/js-name (comp pg/camel-case name)})
+                        response (<? (fetch/request-async (merge
```
Comment (wilkerlucio) [inline src/com/wsscode/pathom/fulcro/network.cljs]: I have another idea on how to extend this, the merging is nice but that blocks using these options to change something else about the process in the future, what if instead the user can provide some fn that will update the request, this way the user could also read the defaults, this merge would be turned in:

```clojure
(cond-> {::http/url url ...} update-http-request update-http-request)
```

Then the user can do:

```clojure
(pfn/graphql-network url {::pfn/update-http-request #(assoc :headers {"myheader" "value"})}
```

What you think?

Comment (MageMasher) [reply src/com/wsscode/pathom/fulcro/network.cljs]: I like it more. Refactoring.

**Comment by MageMasher** [comment]
@wilkerlucio Let me know what you think

**Comment by wilkerlucio** [comment]
Thanks!

**Pull request merged**